app/questionnaire/questionnaire_schema.py

```diff
@@ -14,73 +15,97 @@ def __init__(self, questionnaire_json, language_code=DEFAULT_LANGUAGE_CODE):
         self.language_code = language_code
         self._parse_schema()
 
-    @property
-    def sections(self):
+    def get_sections(self):
         return self._sections_by_id.values()
 
-    @property
-    def groups(self):
+    def get_section_via_section_id(self, section_id: str):
+        return self._sections_by_id.get(section_id)
+
+    def get_section_via_group_id(self, group_id: str):
+        group = self.group(group_id)
+
+        try:
+            return self.get_section_via_section_id(group['parent_id'])
+        except KeyError:
+            return None
+
+    def get_section_via_block_id(self, block_id: str):
+        block = self.block(block_id)
+
+        try:
+            group_id = block['parent_id']
+        except TypeError:
+            return None
+
+        try:
+            section_id = self.group(group_id)['parent_id']
+        except TypeError:
+            section_id = self._get_section_id_via_list_block(block)
+
+        return self.get_section_via_section_id(section_id)
+
+    def get_next_section(self, section_id: str):
+        section_ids = tuple(self._sections_by_id.keys())
+        try:
+            next_section_id = section_ids[section_ids.index(section_id) + 1]
+        except IndexError:
+            return None
+
+        return self.get_section_via_section_id(next_section_id)
+
+    def get_section_summary_by_section_id(self, section_id: str) -> Union[str, None]:
+        section: Mapping = self.get_section_via_section_id(section_id)
+        for group in section.get('groups'):
+            for block in group.get('blocks'):
+                if block['type'] == 'SectionSummary':
+                    return block['id']
+
+    def get_groups(self):
         return self._groups_by_id.values()
 
-    @property
+    def group(self, group_id=None, block_id=None):
+        if group_id:
+            return self._groups_by_id.get(group_id)
+
+        if block_id:
+            return self._group_via_block(block_id)
+
+    def first_block_id_for_group(self, group_id):
+        group = self.group(group_id)
+        if group:
+            return group['blocks'][0]['id']
+
     def blocks(self):
         return self._blocks_by_id.values()
 
-    @property
-    def answers(self):
-        return self._answers_by_id.values()
+    def block(self, block_id=None, answer_id=None):
```

Comment:
Noun type names - i.e blocks, groups, sections should really be properties of a class, though given the method behaviour hasn't changed (and is really doing some work as a look up) should probably be as they were in the original methods.

Reply:
I'd prefer this to be two methods - 1. get_block_by_block_id and 2. get_block_by_answer_id - does that sound good?

Reply:
I think that a method named get_block is going to expect block_id as an arg, and we generally use 'for' to indicate something different - so it probably should just be get_block, get_block_for_answer_id if they're being broken apart. That applies to other accessors too, so get_section_for_section_id could be shortened to just get_section, for instance.

We probably should agree a style guide on the team so we're all happy about the style we're using going forward so we can minimise some of this.

Reply:
Ok, in that case, are you fine for me to leave it as is for now, and update following a style guide upddate?

Reply:
Given it's such a simple change highlighted here, I think it should probably be addressed as part of this PR.